src/packet/mod.rs

```diff
@@ -32,26 +29,43 @@ impl fmt::Display for Packet {
 }
 
 impl Packet {
-    // MarshalSize returns the size of the packet once marshaled.
-    pub fn size(&self) -> usize {
-        self.header.size() + self.payload.len()
+    pub fn new() -> Self {
+        Packet::default()
+    }
+
+    /// Unmarshal parses the passed byte slice and stores the result in the Header this method is called upon
+    pub fn unmarshal(&mut self, buf: &mut [u8]) -> Result<(), RTPError> {
```

Comment:
another general issue about unmarshal signature.

it is preferred that  buf is read-only for unmarshal:
pub fn unmarshal(&mut self, buf: &[u8])

Reply:
Sure, this should be done as buf isn't even mutated

Reply:
`packet.unmarshal` calls `header.unmarshal` which is a mutable

Reply:
Missed this. Updated codebase, thanks.